src/lib/pages/frontsite/Series/SeriesDetails/SeriesPanel/SeriesPanel.js

```diff
@@ -34,10 +35,9 @@ class SeriesPanel extends Component {
                 </Grid.Column>
                 <Grid.Column width={6}>
                   <ILSHeaderPlaceholder isLoading={isLoading}>
-                    <SeriesTitle
-                      title={series.metadata.title}
-                      modeOfIssuance={series.metadata.mode_of_issuance}
-                    />
+                    {title && moi && (
```

Comment:
as the moi is required, the title too, you could just do 
```
isEmpty(series.metadata) && (
```

Reply:
yes but this check will not fail if you have one of the two, since it will not be empty. To be honest I had it `!isLoading` as a condition for its render, but switched it to explicit `title` and `moi` so you don't have to dive further to check what is required since its already there. No really strong opinion.

Reply:
not sure I understand, they are both required no? What is the issue here?

Reply:
If we don't prevent the component from rendering when they don't exist we get errors, I believe this is the way to use the `required` propTypes.

Reply:
yes perfect, agree, but `title` and `moi` are mandatory in the JSON schema, how come they are not defined?

Reply:
@ntarocco our [`initialState`](https://github.com/inveniosoftware/react-invenio-app-ils/blob/4aede755893278150d0ec856716841259e2a41b5/src/lib/pages/frontsite/Series/SeriesDetails/state/reducer.js#L6) in the reducer is `data: { metadata: {} }`, and goes `SeriesDetail` -> `SeriesPanel` -> `SeriesTitle`.

We had something like the following 

`<SeriesTitle title={series.metadata.title} modeOfIssuance={series.metadata.mode_of_issuance} />`

which would spit warnings like

```
Warning: Failed prop type: The prop `title` is marked as required in `SeriesTitle`, but its value is `undefined`.
```

Reply:
yes, is this short moment when we do not have yet the data and the components use the initial state from the reducer
one of the solutions could be to provide the initial object with the required fields, but it seems problematic to maintain (in case of changes we need to remember to go and change the initial state too), however it seems safer to fail on the initial state